### DISCUSSION TITLE
Autodispose provider is disposing when doing pumpAndSettle when testing

### DISCUSSION THREAD

**Discussion by MisterLight** [rrousselGit/riverpod#2749] [Q&A]
I have a widget that use a provider for storing data. In my widget I trigger a notification, get value from the notification, and save it on the provider, later a field is set with this value that read from the provider.

 In my tests Im trying to replicate this functionality but when I trigger pumpAndSettle (to hide the notification popup), the provider is disposed.  ChatGPT told me that pumpAndSettle() trigger the dispose in autoDispose providers but don't give me a clear solution. Maybe keepAlive will do the trick? How can I solve this?, I don't find information about this issue, Im the only one?

Workflow looks like this:

In my widget
   - ref.read(notificationProvider.notifier).setNotificationId('theId');
   
In my tests

- pumpAndSettle() //  trigger dispose method

In my widget again

- String id = ref.read(notificationProvider.notifier).notificationId // When read, the provider is created again. NotificationID is null.
- test fail

**Comment by rrousselGit** [comment]
Providers that aren't listened will be disposed.
If you want your provider to not get disposed, you should listen to them somewhere
